src/librustc_mir/transform/qualify_consts.rs

```diff
@@ -569,6 +470,8 @@ impl<'a, 'tcx> Visitor<'tcx> for Qualifier<'a, 'tcx, 'tcx> {
                             }
                         }
 
+                        ProjectionElem::ConstantIndex {..} |
+                        ProjectionElem::Subslice {..} |
```

Comment:
Same here

Reply:
This is truly a bugfix. When we stabilized/implemented slice patterns, we forgot to also make them work for const fn arguments